pavelib/quality.py

```diff
@@ -301,47 +304,140 @@ def run_jshint(options):
-                safelint_report=safelint_report
+            "Error. Number of {safelint_script} violations could not be found in {safelint_report}".format(
+                safelint_script=safelint_script, safelint_report=safelint_report
             )
         )
 
+    # Print number of violations to log.
+    violations_count_str = "Number of {safelint_script} violations: {num_violations}".format(
+        safelint_script=safelint_script, num_violations=num_violations
+    )
+    print violations_count_str
+
     # Record the metric
-    _write_metric(num_violations, (Env.METRICS_DIR / "safelint"))
+    _write_metric(violations_count_str, (Env.METRICS_DIR / "safelint"))
 
-    # Fail if number of violations is greater than the limit
-    if num_violations > violations_limit > -1:
-        raise Exception(
-            "SafeTemplateLinter Failed. Too many violations ({count}).\nThe limit is {violations_limit}.".format(
-                count=num_violations, violations_limit=violations_limit
+    error_message = ""
+
+    # Test total violations against threshold.
+    if 'total' in violation_thresholds.keys():
+        if violation_thresholds['total'] < safelint_counts['total']:
+            error_message = "Too many violations total ({count}).\nThe limit is {violations_limit}.".format(
+                count=safelint_counts['total'], violations_limit=violation_thresholds['total']
+            )
+
+    # Test rule violations against thresholds.
+    if 'rules' in violation_thresholds:
+        threshold_keys = sorted(violation_thresholds['rules'].keys())
+        for threshold_key in threshold_keys:
+            if threshold_key not in safelint_counts['rules']:
+                error_message += (
+                    "\nNumber of {safelint_script} violations for {rule} could not be found in "
+                    "{safelint_report}."
+                ).format(
+                    safelint_script=safelint_script, rule=threshold_key, safelint_report=safelint_report
+                )
+            elif violation_thresholds['rules'][threshold_key] < safelint_counts['rules'][threshold_key]:
+                error_message += \
+                    "\nToo many {rule} violations ({count}).\nThe {rule} limit is {violations_limit}.".format(
+                        rule=threshold_key, count=safelint_counts['rules'][threshold_key],
+                        violations_limit=violation_thresholds['rules'][threshold_key],
+                    )
+
+    if error_message is not "":
+        raise BuildFailure(
+            "SafeTemplateLinter Failed.\n{error_message}\n"
+            "See {safelint_report} or run the following command to hone in on the problem:\n"
+            "  ./scripts/safe-commit-linter.sh -h".format(
+                error_message=error_message, safelint_report=safelint_report
+            )
+        )
+
+
+@task
+@needs('pavelib.prereqs.install_python_prereqs')
+def run_safecommit():
```

Comment:
No, I was just thinking of the newbie that might be seeing these commands on paver. I wouldn't want him/her to think they run this paver target to make their commit safe. I was imagining even with `report` it would fail. So, perhaps I need to think it through some more. But my main point is to make it clear that this is a linter.